.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,15 @@
+name: pre-commit
+
+on:
+    pull_request:
+        branches: [main]
+
+jobs:
+    pre-commit:
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v5
+            - uses: actions/setup-python@v6
+              with:
+                  python-version: "3.12"
+            - uses: pre-commit/action@v3.0.1
```

Comment:
I've rolled without using the dedicated pre-commit action before, but something I've added in my own CI is the `--show-diff-on-failure` flag. It runs git diff on a failure so you can see the issue in the readout. Unsure if the CI action is applying it under the hood (should be easy enough to check source for that action and confirm if they're using it or not; they likely also have a way to pass arbitrary flags).